File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/Location.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class Location 
{
    public long Id { get; set; }
    public bool Active { get; set; }
    public bool Valid { get; set; }
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public string LocationDescription { get; set; }
    public string Name { get; set; }
    public Territory Territory { get; set; }
    public ServiceHub Hub { get; set; }
    public Asset Picture { get; set; }
    public List<Note> Notes { get; set; }
    public long NoteCount { get; set; }
    public string HomePhone { get; set; }
    public string CellPhone { get; set; }
    public CellCarrier CellCarrier { get; set; }
    public string BusinessPhone { get; set; }
    public string BusinessPhoneExt { get; set; }
    public string FaxNumber { get; set; }
    public string VarTimeZone { get; set; }
    public string UtcOffset { get; set; }
    public string Code501c3 { get; set; }
    public string EmailAddress { get; set; }
    public Address Address { get; set; }
    public string Web { get; set; }
    public Program Program { get; set; }
    
    /// <summary>
    /// Gets or Sets Visibility
    /// </summary>
    public enum VisibilityEnum
    {
        
        /// <summary>
        /// Enum PUBLICEnum for PUBLIC
        /// </summary>
        PUBLICEnum = 1,
        
        /// <summary>
        /// Enum PRIVATEEnum for PRIVATE
        /// </summary>
        PRIVATEEnum = 2,
        
        /// <summary>
        /// Enum FRIENDSEnum for FRIENDS
        /// </summary>
        FRIENDSEnum = 3
    }

    public VisibilityEnum Visibility { get; set; }
    public string GroupingId { get; set; }
    public bool Destination { get; set; }
    public long DockingTime { get; set; }
    public long UsageCount { get; set; }
    public string TimeFramesAllowed { get; set; }
    public List<YayOrNay> Likes { get; set; }
    public long LikeCount { get; set; }
    public long DislikeCount { get; set; }
    public string OwnerDisplay { get; set; }
    public string ContentName { get; set; }
}

File: server-scala-play-server/app/model/Location.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for Location.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class Location(
  id: Option[Long],
  active: Option[Boolean],
  valid: Option[Boolean],
  latitude: Option[Double],
  longitude: Option[Double],
  locationDescription: Option[String],
  name: Option[String],
  territory: Option[Territory],
  hub: Option[ServiceHub],
  picture: Option[Asset],
  notes: Option[List[Note]],
  noteCount: Option[Long],
  homePhone: Option[String],
  cellPhone: Option[String],
  cellCarrier: Option[CellCarrier],
  businessPhone: Option[String],
  businessPhoneExt: Option[String],
  faxNumber: Option[String],
  timeZone: Option[String],
  utcOffset: Option[String],
  code501c3: Option[String],
  emailAddress: Option[String],
  address: Option[Address],
  web: Option[String],
  program: Option[Program],
  visibility: Option[Location.Visibility.Value],
  groupingId: Option[String],
  destination: Option[Boolean],
  dockingTime: Option[Long],
  usageCount: Option[Long],
  timeFramesAllowed: Option[String],
  likes: Option[List[YayOrNay]],
  likeCount: Option[Long],
  dislikeCount: Option[Long],
  ownerDisplay: Option[String],
  contentName: Option[String]
)

object Location {
  implicit lazy val locationJsonFormat: Format[Location] = Json.format[Location]

  // noinspection TypeAnnotation
  object Visibility extends Enumeration {
    val PUBLIC = Value("PUBLIC")
    val PRIVATE = Value("PRIVATE")
    val FRIENDS = Value("FRIENDS")

    type Visibility = Value
    implicit lazy val VisibilityJsonFormat: Format[Value] = Format(Reads.enumNameReads(this), Writes.enumNameWrites[this.type])
  }
}

